src/hooks/useDebounce.js

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useEffect, useRef } from 'react';
+
+const useDebounce = (handler, dependency = [], time = 300) => {
+  const $timer = useRef(null);
+
+  useEffect(() => {
+    if ($timer.current != null) {
+      clearTimeout($timer.current);
+    }
+
+    $timer.current = setTimeout(() => {
+      handler();
+    }, time);
+
+    return () => {
+      clearTimeout($timer.current);
+    };
+  }, dependency);
+};
+
+export default useDebounce;
```

Comment:
Very good use of hook.

The eslint rule react-hooks/exhaustive-deps exists for one reason, it can alert us for possible bugs or unexpected behaviors on the app.

Please remove this ignore rule comment and try to fix the eslint message.